### ISSUE TITLE
Update to Golden-Layout v2

### ISSUE THREAD

**Issue by RubenRBS** [compiler-explorer/compiler-explorer#4175]
We're stuck in the latest mejor 1.5.9 version. 

Ideally the switch would not be too hard once we completely switch to TS, and it would fix some typing issues we have had

**Comment by mattgodbolt** [comment]
We have to be super careful as effectively our URL parameter schema is tied into the specifics of GL's window layout

**Comment by mattgodbolt-molty** [comment]
Two things get conflated whenever this comes up, and only one of them is this issue. I went looking to see whether we should branch our own copy of the v1 we actually run, purely to take security and other patches, and the answer turns out to be that we already did.

**The cheap thing, already done.** Since June 2026 `package.json` points at `github:compiler-explorer/golden-layout#v1.6.0` (ef77b8b here, lock pins `e921d08`), which is our own fork on the `ce-main` branch. It is 10 commits ahead of upstream v1.5.9 and 0 behind. It carries a modernised build toolchain, a GitHub Actions CI run, and an XSS hardening fix that stops a header button's label/class and the pop-in button's title being concatenated into an HTML string, with tests (compiler-explorer/golden-layout#1 and #2). Credit to @partouf.

**What that does and does not buy us.** There is no upstream v1 stream left to sync from. Upstream's last v1 release was 1.5.9 in July 2017. `master` and `dev` are the v2 line: the last functional change landed in July 2022 and shipped as v2.6.0 that September, and everything since is documentation and lockfile churn, plus a couple of unreleased drag fixes on `dev` in June 2024. So "get it up to date with upstream security patches" is not an action available to us. Our fork is the patch stream. Anything we want fixed in v1 we fix ourselves, and nobody upstream is going to hand it to us. That is still much cheaper than this issue, but the maintenance is ours.

**On advisories, so nobody reads more into this than there is: I found none.** GitHub's advisory database returns nothing for the npm package `golden-layout`, OSV returns nothing, and `npm audit` on our tree currently reports 10 findings (4 high, 6 moderate) across hono, undici, dompurify, brace-expansion, ip-address, protobufjs and fast-uri, none of which are golden-layout or jquery. The XSS fix above was hardening we chose to do, not a response to a published advisory. Worth noting that we are on jQuery 4.0.0 while golden-layout declares `jquery: "*"`, so nothing but our own testing pins that pairing.

One real cost of the fork as it stands is that it is a git dependency rather than a published package, which is what #8952 is about (npm 12 defaults `--allow-git` to none). If that gets annoying the options are publishing it to npm under our scope or vendoring the built files. Not urgent, just a tradeoff we have taken.

**The expensive thing, which is this issue.** v2 is an API rewrite, not a version bump. Today golden-layout is referenced 372 times across 77 files, 52 of them under `static/`, with 45 direct imports. Every pane is built on the v1 container and state model, `static/hub.ts` walks `ContentItem` trees and reaches into `container.tab.header.parent`, and `static/components.interfaces.ts` extends `GoldenLayout.Config` directly.

The code volume is not the hard part. The hard part is still what I said in 2022: our URL schema *is* GL's layout config, and we have since built more on top of that, not less. `shared/url-serialization.ts` now contains our own reimplementation of GL's ConfigMinifier so the server can handle configs without a browser, and `lib/clientstate-normalizer.ts` reads the same shape. Any move to v2 has to keep every existing short link working, which realistically means a versioned conversion layer rather than a swap. The type-safety work in #7807, #7808 and #7843 is the useful groundwork for that.

My view: we stay on the fork, treat it as ours to patch rather than something we are waiting on upstream for, and keep this issue for the v2 work so the two do not blur together. I am not proposing we start the migration.

(Verified: the package.json/lock pin, the fork's branch and commit contents, upstream's tag and commit history, the three advisory sources, and the reference counts. Not verified: whether jQuery 4 has any behavioural incompatibility with GL 1.6.0 beyond what our tests already exercise, and I have not attempted to estimate the v2 migration in person-time.)
